data_source.go

```diff
@@ -337,6 +338,27 @@ func (ds *AnySource) PrepareRun() error {
 	ds.runDone.Add(ds.nchan)
 	signed := ds.Signed()
 	vpa := ds.VoltsPerArb()
+
+	// Load last trigger state from config file
+	var fts []FullTriggerState
+	if err := viper.UnmarshalKey("trigger", &fts); err != nil {
+		// could not read trigger state from config file.
+		fts = []FullTriggerState{}
+	}
+	tsptrs := make([]*TriggerState, ds.nchan)
+	for i, ts := range fts {
+		for _, chnum := range ts.ChanNumbers {
+			if chnum < ds.nchan {
+				tsptrs[chnum] = &(fts[i].TriggerState)
+			}
+		}
+	}
+	// Use defaultTS for any channels not in the stored state.
+	// This will be needed any time you have more channels than in the
+	// last saved configuration.
+	defaultTS := TriggerState{AutoDelay: 250 * time.Millisecond,
```

Comment:
This should be all OFF! It's easy to turn them on, it's super frustrating to track down why a channel is triggering when I don't expect it to do so.

Reply:
This is OFF. But I can be more explicit, if you like.